### PR TITLE
Adds config class for ChallengeSession

### PR DESCRIPTION
**Pull request by CarolineClark**

### CODE REVIEW

File: src/main/java/tdl/client/runner/ChallengeSession.java [hunk 1]
```diff
@@ -13,57 +12,27 @@
...
-
-    public ChallengeSession withImplementationRunner(ImplementationRunner implementationRunner) {
-        this.implementationRunner = implementationRunner;
-        return this;
-    }
-
-    public ChallengeSession withRecordingSystemOn(boolean recordingSystemOn) {
-        this.recordingSystem = new RecordingSystem(recordingSystemOn);
+    public ChallengeSession withConfig(ChallengeSessionConfig config) {
+        this.hostname = config.getHostname();
```
Comment (julianghionoiu) [inline src/main/java/tdl/client/runner/ChallengeSession.java]: We just need to store the config itself and remove the other variables.

Comment (CarolineClark) [reply src/main/java/tdl/client/runner/ChallengeSession.java]: Who owns the RecordingSystem member variable?

Comment (julianghionoiu) [reply src/main/java/tdl/client/runner/ChallengeSession.java]: `ChallengeSession`

**Review by julianghionoiu — COMMENTED** [review]
Looks good.
If we introduce the same config concept for the ImplementationRunner then we are good to merge.

**Comment by CarolineClark** [comment]
Ah, I didn't see the implementationConfig - that part of the code snippet was not visible. Doing now.

**Review by julianghionoiu — APPROVED** [review]

**Pull request merged**